src/home/useGroupCallRooms.ts

```diff
@@ -89,18 +89,44 @@ const roomIsJoinable = (room: Room): boolean => {
     return false;
   }
   // otherwise we can always join rooms because we will automatically decide if we want to use perParticipant or password
-  const joinRule = room.getJoinRule();
-  return joinRule === JoinRule.Knock || joinRule === JoinRule.Public;
+  switch (room.getJoinRule()) {
+    case JoinRule.Public:
+      return true;
+    case JoinRule.Knock:
+      switch (room.getMyMembership()) {
+        case KnownMembership.Join:
+        case KnownMembership.Knock:
+          return true;
+        case KnownMembership.Invite:
+          return (
+            room
+              .getLiveTimeline()
+              .getState(EventTimeline.FORWARDS)
+              ?.getStateEvents(EventType.RoomMember, room.myUserId)
+              ?.getPrevContent().membership === JoinRule.Knock
+          );
```

Comment:
I am wondering if a normal invite should also be in the room list.
In case I am inviting a guest user from callA to a new callB.

But then this would break the semantic of recents...

So I think were you are coming from and if you prefer to stick to what one would expect from recents but as a consequence breaking the usecase outlined above I am all for it.

(The above would just need to be handled like so: invite geust to callB send guest invite link -> with you additions that would result in an immediate join)

Reply:
I agree on both points: it'd be useful to see normal invites, but not as an item in the recents list.

What's more, ECall tends to try to auto-join rooms, so showing general-invite rooms in the recents list would maybe auto-accept them if we don't add an exception for them.

As a compromise, we could add a dedicated section for "Invites" separate from the recents list.  Though that's something a new issue/PR should address.